cmd/snap/cmd_advise.go

```diff
@@ -90,11 +100,122 @@ func outputAdviseJSON(command string, results []advisor.Command) error {
 	return nil
 }
 
+type jsonRPC struct {
+	JsonRPC string `json:"jsonrpc"`
+	Method  string `json:"method"`
+	Params  struct {
+		Command         string   `json:"command"`
+		SearchTerms     []string `json:"search-terms"`
+		UnknownPackages []string `json:"unknown-packages"`
+		Packages        []string `json:"packages"`
+	} `json:"params"`
+}
+
+func readRpc(r *bufio.Reader) (*jsonRPC, error) {
+	line, _, err := r.ReadLine()
+	if err != nil {
+		return nil, fmt.Errorf("cannot read json-rpc: %v", err)
+	}
+
+	var rpc jsonRPC
+	if err := json.Unmarshal(line, &rpc); err != nil {
+		return nil, err
+	}
+	// empty \n
+	emptyNL, _, err := r.ReadLine()
+	if err != nil {
+		return nil, err
+	}
+	if string(emptyNL) != "" {
+		return nil, fmt.Errorf("unexpected line: %q (empty)", emptyNL)
+	}
+
+	return &rpc, nil
+}
+
+func adviceViaAptHook() error {
+	sockFd := os.Getenv("APT_HOOK_SOCKET")
+	if sockFd == "" {
+		return fmt.Errorf("cannot find APT_HOOK_SOCKET env")
```

Comment:
Should we just return nil here? Will this be the case when older apt is used and there's just no socket RPC to do.

Reply:
Apt added support for the new kind of hooks at the same time as the json rpc was added. So if there is no APT_HOOK_SOCKET env but the hook is called something is wrong (on the apt side).